x-pack/plugins/ml/public/application/jobs/jobs_list/components/job_details/extract_job_details.js

```diff
@@ -11,12 +11,15 @@ import { formatValues, filterObjects } from './format_values';
 import { i18n } from '@kbn/i18n';
 import { EuiLink } from '@elastic/eui';
 import { EditAlertRule } from '../../../../../alerting/ml_alerting_flyout';
+import { removeNodeInfo } from '../../../../../../common/util/job_utils';
 
-export function extractJobDetails(job, basePath, refreshJobList) {
-  if (Object.keys(job).length === 0) {
+export function extractJobDetails(originalJob, basePath, refreshJobList, isServerless) {
+  if (Object.keys(originalJob).length === 0) {
     return {};
   }
 
+  const job = isServerless ? removeNodeInfo(originalJob) : originalJob;
```

Comment:
❓ Should we remove this data on the Kibana server side instead? 
All the checks below could be removed then.

Reply:
I've changed this so the node information is removed just after it is loaded for the job lists. So the `isServerless` flag no longer need to be passed to this component.